src/runtime/library_module.cc

```diff
@@ -123,20 +126,63 @@ void ImportModuleBlob(const char* mblob, std::vector<Module>* mlist) {
   dmlc::Stream* stream = &fs;
   uint64_t size;
   CHECK(stream->Read(&size));
+  std::vector<Module> modules;
+  bool has_import_tree = false;
   for (uint64_t i = 0; i < size; ++i) {
     std::string tkey;
     CHECK(stream->Read(&tkey));
-    if (tkey == "c") continue;
-    std::string fkey = "module.loadbinary_" + tkey;
-    const PackedFunc* f = Registry::Get(fkey);
-    CHECK(f != nullptr)
+    // Currently, _lib is for DSOModule, but we
+    // don't have loadbinary function for it currently
+    if (tkey == "_lib") {
+      auto dso_module = Module(make_object<LibraryModuleNode>(lib));
+      // allow lookup of symbol from dso root (so all symbols are visible).
+      if (auto *ctx_addr =
+          reinterpret_cast<void**>(lib->GetSymbol(runtime::symbol::tvm_module_ctx))) {
+        *ctx_addr = dso_module.operator->();
+      }
+      modules.emplace_back(dso_module);
+    } else if (tkey == "_import_tree") {
+      has_import_tree = true;
+      std::vector<uint64_t> import_tree_row_ptr;
```

Comment:
Move the importation logic outside the load loop. This way the import tree can also be in the middle.

Reply:
Could you what is the meaning of move the logic a bit more? Not understand fully.

Reply:
move ```      std::vector<uint64_t> import_tree_row_ptr;``` and the indices outside the loop. In this scope, only load these data structures, but do not perform the import(L150) move the import logic after the for loop

Reply:
ok. Got it.